### ISSUE TITLE
[ERROR] Unable to parse Advanced Threat Protection Set ID

### ISSUE THREAD

**Issue by odee30** [terraform-providers/terraform-provider-azurerm#10214]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Terraform v0.14.2
Provider v2.32.0

### Affected Resource(s)

azurerm_advanced_threat_protection

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "example" {
  name                     = "atptestdevuks"
  location                 = azurerm_resource_group.example.location
  resource_group_name      = azurerm_resource_group.example.name
  account_tier             = "Standard"
  account_replication_type = "LRS"

  tags = module.tagging.tags
}

resource "azurerm_advanced_threat_protection" "example" {
  target_resource_id = azurerm_storage_account.example.id
  enabled            = true
}
```

### Expected Behaviour

Terraform plan and apply should succeed and advanced threat protection should be enabled for the storage account.

### Actual Behaviour

Plan fails with the error:
[ERROR] Unable to parse Advanced Threat Protection Set ID: subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/example-dev-uks-rg/providers/Microsoft.Storage/storageAccounts/atptestdevuks/providers/Microsoft.Security/advancedThreatProtectionSettings/current: invalid URI for request

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Looking at the uri closer and in state it looks as if it is missing at '/' at the start

### References

* #0000

**Comment by odee30** [comment]
I see this has already been raised in #5623

**Comment by Wrighty11146** [comment]
Hi I seem to be facing the same error, 

It seems that ATP is not parsing the beginning '/' on the terraform plan/apply it is showing but then doesnt go through, if you go into the state file, you can add it in there as a work around but we shouldnt need to do this. 

Any advise on how this could be resolved.

**Comment by odee30** [comment]
> Hi I seem to be facing the same error,
> 
> It seems that ATP is not parsing the beginning '/' on the terraform plan/apply it is showing but then doesnt go through, if you go into the state file, you can add it in there as a work around but we shouldnt need to do this.
> 
> Any advise on how this could be resolved.

Upgrading the provider to v2.42 seemed to do the trick for me.

**Comment by Tbohunek** [comment]
I also confirm it works with v2.43.

**Comment by pearcec** [comment]
> Upgrading the provider to v2.42 seemed to do the trick for me.

The fix was introduced in v2.43.0.  It (v2.42.0) did not fix it for me.
